core/src/main/java/de/danoeh/antennapod/core/preferences/UserPreferences.java

```diff
@@ -1046,14 +1045,14 @@ public static void setQueueKeepSortedOrder(SortOrder sortOrder) {
                 .apply();
     }
 
-    public static int getFeedFilter() {
-        String value = prefs.getString(PREF_FILTER_FEED, "" + FEED_FILTER_NONE);
-        return Integer.parseInt(value);
+    public static SubscriptionsFilter getSubscriptionsFilter() {
+        String value = prefs.getString(PREF_FILTER_FEED, "");
+        return new SubscriptionsFilter(value);
     }
 
-    public static void setFeedFilter(String value) {
+    public static void setSubscriptionsFilter(SubscriptionsFilter value) {
         prefs.edit()
-                .putString(PREF_FILTER_FEED, value)
+                .putString(PREF_FILTER_FEED, TextUtils.join(",", value.getValues()))
```

Comment:
I think a method like `value.serialize()` would be better. This then keeps the details of how items are joined encapsulated within the class.